ext/Pod-Html/lib/Pod/Html.pm

```diff
@@ -239,237 +238,234 @@ This program is distributed under the Artistic License.
     if (defined $opts->{flush}) {
-        1 while unlink($globals->{Dircache});
+        1 while unlink($self->{Dircache});
     }
-    return $globals;
+    return $self;
 }
 
 sub refine_globals {
-    my $globals = shift;
-    require Data::Dumper if $globals->{verbose};
+    my $self = shift;
+    require Data::Dumper if $self->{verbose};
 
     # prevent '//' in urls
-    $globals->{Htmlroot} = "" if $globals->{Htmlroot} eq "/";
-    $globals->{Htmldir} =~ s#/\z##;
+    $self->{Htmlroot} = "" if $self->{Htmlroot} eq "/";
+    $self->{Htmldir} =~ s#/\z##;
 
-    if (  $globals->{Htmlroot} eq ''
-       && defined( $globals->{Htmldir} )
-       && $globals->{Htmldir} ne ''
-       && substr( $globals->{Htmlfile}, 0, length( $globals->{Htmldir} ) ) eq $globals->{Htmldir}
+    if (  $self->{Htmlroot} eq ''
+       && defined( $self->{Htmldir} )
+       && $self->{Htmldir} ne ''
+       && substr( $self->{Htmlfile}, 0, length( $self->{Htmldir} ) ) eq $self->{Htmldir}
        ) {
         # Set the 'base' url for this file, so that we can use it
         # as the location from which to calculate relative links
         # to other files. If this is '', then absolute links will
         # be used throughout.
-        #$globals->{Htmlfileurl} = "$globals->{Htmldir}/" . substr( $globals->{Htmlfile}, length( $globals->{Htmldir} ) + 1);
-        # Is the above not just "$globals->{Htmlfileurl} = $globals->{Htmlfile}"?
-        $globals->{Htmlfileurl} = unixify($globals->{Htmlfile});
-
+        #$self->{Htmlfileurl} = "$self->{Htmldir}/" . substr( $self->{Htmlfile}, length( $self->{Htmldir} ) + 1);
+        # Is the above not just "$self->{Htmlfileurl} = $self->{Htmlfile}"?
+        $self->{Htmlfileurl} = unixify($self->{Htmlfile});
     }
-    return $globals;
+    return { %{$self} };
 }
 
 sub generate_cache {
-    my ($globals, $Pagesref) = @_;
+    my $self = shift;
     my $pwd = getcwd();
-    chdir($globals->{Podroot}) ||
-        die "$0: error changing to directory $globals->{Podroot}: $!\n";
+    chdir($self->{Podroot}) ||
+        die "$0: error changing to directory $self->{Podroot}: $!\n";
 
     # find all pod modules/pages in podpath, store in %Pages
     # - inc(0): do not prepend directories in @INC to search list;
-    #     limit search to those in @{$globals->{Podpath}}
+    #     limit search to those in @{$self->{Podpath}}
     # - verbose: report (via 'warn') what search is doing
     # - laborious: to allow '.' in dirnames (e.g., /usr/share/perl/5.14.1)
-    # - callback: used to remove Podroot and extension from each file
     # - recurse: go into subdirectories
     # - survey: search for POD files in PodPath
     my ($name2path, $path2name) = 
-        Pod::Simple::Search->new->inc(0)->verbose($globals->{Verbose})->laborious(1)
-        ->callback(\&_save_page)->recurse($globals->{Recurse})->survey(@{$globals->{Podpath}});
-    #print STDERR Data::Dumper::Dumper($name2path, $path2name) if ($globals->{Verbose});
```

Comment:
This comment gets removed but seems to be the only reason Data::Dumper is required when the verbose option is passed. Maybe that could be removed as well.